### PR TITLE
Wagmanbe/eam/waccm e90 st80

### PR DESCRIPTION
**Pull request by wagmanbe**
This pull request contains my branch which is a duplicate of Allen Hu's branch. The only difference is that I have updated the chemistry pre-processor file `components/eam/src/chemistry/pp_linoz_mam4_resus_mom_soag/chem_mech.in` to include ST80 based on Allen's most recent file, which he emailed to me. I have not confirmed this runs, nor do I know what, if anything, is necessary to add to the runscript to make it run. I will ask Allen to provide those details, if they are necessary.

### CODE REVIEW

**Comment by wagmanbe** [comment]
I force pushed a commit that reverts chem_mech.in back to the default E3SM chem_mech.in. This makes my current branch identical to Allen's branch. User must now supply chem_mech.in which includes e90 and st80 in the runscript using, e.g  Hunter and Allen's style: 
`# Specify a chem_mech.in file. 
export usr_mech_infile=/projects/cldera/data/E3SM/model_input/e90st80/chem_mech_bothtracers.in  ##<<<<<<<<< define the new chem_mech.in file
` 
and then in case.setup
`./xmlchange --id CAM_CONFIG_OPTS --append --val=' -usr_mech_infile $usr_mech_infile'`

**Comment by wagmanbe** [comment]
My latest commit uses git rebase to apply Allen's e90/st80 on top of the master branch (which now includes Hunter's prognostic aerosol. 
The runscript is [here](https://gitlab-ex.sandia.gov/cldera/simulated-pathways/-/blob/master/E3SM_runscripts/F20TR/presc/run.v2.ULR.coldstart.presc.e90st80.configflag.sh)

File: components/eam/bld/build-namelist [hunk 1]
```diff
@@ -2875,10 +2875,11 @@ if ($chem eq 'linoz_mam4_resus_mom' || $chem eq 'linoz_mam4_resus_mom_soag') {
 
     my %species;
 
-    # Surface emission datasets:
+    # Surface emission datasets (AH -- added E90 here):
```
Comment (brhillman) [inline components/eam/bld/build-namelist:2878]: I would remove the signatures, we can always find diffs using git diff, and attribute changes via git blame.

File: components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90 [hunk 2]
```diff
@@ -27,6 +27,9 @@ module mo_gas_phase_chemdr
   integer :: ndx_cldfr, ndx_cmfdqr, ndx_nevapr, ndx_cldtop, ndx_prain, ndx_sadsulf
   integer :: ndx_h2so4
   integer :: inv_ndx_cnst_o3, inv_ndx_m
+!AH
```
Comment (brhillman) [inline components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90:30]: I'd remove signature/initials

File: components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90 [hunk 3]
```diff
@@ -63,6 +66,9 @@ subroutine gas_phase_chemdr_inti(chem_name)
          convproc_do_aer_out = convproc_do_aer ) 
    
     ndx_h2so4 = get_spc_ndx('H2SO4')
+!AH
```
Comment (brhillman) [inline components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90:69]: Remove signatures/initials; replace with more meaningful documentation if necessary.

File: components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90 [hunk 4]
```diff
@@ -449,7 +455,14 @@ subroutine gas_phase_chemdr(lchnk, ncol, imozart, q, &
     !-----------------------------------------------------------------------      
     !        ... Xform from mmr to vmr
     !-----------------------------------------------------------------------      
-    call mmr2vmr( mmr, vmr, mbar, ncol )
+!    call mmr2vmr( mmr, vmr, mbar, ncol )
+!AH -- added ST80
```
Comment (brhillman) [inline components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90:459]: Remove signatures/initials; replace with more meaningful documentation if necessary. Might be worthwhile to put in a comment explaining what this is and why.

File: components/eam/src/chemistry/pp_linoz_mam4_resus_mom_soag/chem_mech.in [hunk 5]
```diff
@@ -1,6 +1,7 @@
    SPECIES
-
+*AH -- added E90
```
Comment (brhillman) [inline components/eam/src/chemistry/pp_linoz_mam4_resus_mom_soag/chem_mech.in:2]: Remove initials.

File: components/eam/src/chemistry/pp_linoz_mam4_resus_mom_soag/chem_mech.in [hunk 6]
```diff
@@ -55,14 +57,15 @@
       Photolysis
  [jh2o2]    H2O2 + hv ->
       End Photolysis
-
+*AH -- added E90
```
Comment (brhillman) [inline components/eam/src/chemistry/pp_linoz_mam4_resus_mom_soag/chem_mech.in:60]: Remove initials

**Comment by wagmanbe** [comment]
> So, does this just add the _ability_ to run with e90/st80, but the user-mods in the script are required to actually run them? And in that case, should a normal run be unchanged?

You raise a good point. The key line in the runscript is: `./xmlchange --id CAM_CONFIG_OPTS --append --val=' -usr_mech_infile $usr_mech_infile'` If the user does not include this, there might be a problem because the other two files (build-namelist and mo_gas_phase_chemdir) expect e90 and st80. Also, there is no need for a modified chem_mech.in in this pull request. I should revert that back to the default file. It doesn't do anything here--it doesn't even get read.

**Comment by wagmanbe** [comment]
I think that the solution is to use the flag that Joe or Hunter is using to turn on CLDERA strat volc or CLDERA passive tracers. If that flag is true, then 1) provide  -usr_mech_infile $usr_mech_infile flag 2) Include E90 and ST80 in mo_gas_phase_chemdr.F90 and build-namelist.

**Comment by wagmanbe** [comment]
Looking at the files that changed (50!) This is artificially large because I actually deleted all the source code from pp_linoz_mam4_resus_soag by mistake, and then checked it back out. For some reason it still shows those files as changed, but I don't think they actually are?

**Comment by brhillman** [comment]
Closing this for now in favor of a lighter-weight solution. Does not currently build or run in the state it is in, will need additional debugging and changes if we want to return to this. Keeping as a branch for now in case we want to go back to this approach.

**Pull request closed without merging**